src/helpers/augmentedReality/multipleSceneIndexGenerator.ts

```diff
@@ -47,14 +73,58 @@ export const multipleSceneIndexGenerator = ({
     const texture = Math.floor(currentDayNumber / timePeriodInDays);
     const model = Math.floor(texture / variableTexturesCount);
     modelIndex = model % scenesCount;
-    const textureIndex = texture % variableTexturesCount;
-    const variableTexture = variableTextures?.[textureIndex];
-    localUris = scenes?.[modelIndex]?.localUris;
 
-    _remove(localUris, ({ type, stable }) => type === 'texture' && !stable);
+    // When `variableTextures` are created, they are created according to index 0 of the scene array.
+    // Once the `modelIndex` is found, it must be updated to select the correct texture file.
+    if (modelIndex > 0) {
+      variableTextures = scenes[modelIndex]?.localUris?.filter(
+        ({ type, stable }) => type === 'texture' && !stable
+      );
+    }
 
-    !!variableTexture && localUris.push(variableTexture);
+    localUris = scenes?.[modelIndex]?.localUris;
+    const textureIndex = texture % variableTexturesCount;
+    textures.push(variableTextures?.[textureIndex] as TTextures[0]);
   }
 
-  return { localUris };
+  for (let i = 0; i < scenes.length; i++) {
+    if (scenesCount > 1 && startDate && timePeriodInDays) {
+      // INFO: localUris = scenes?.[modelIndex]?.localUris;
+      // `localUri` is generated according to `modelIndex`.
+      localUris.forEach((subItem) => {
+        modelsAndTexturesArrayGenerator(subItem, models, textures, true);
+      });
+      break;
+    } else {
+      localUris = scenes?.[i]?.localUris;
+      localUris.forEach((subItem) => {
+        modelsAndTexturesArrayGenerator(subItem, models, textures);
+      });
+    }
+  }
```

Comment:
i think this can be optimized to reduce repetition and conditions:

```
for (let i = 0; i < scenes.length; i++) {
  if (scenesCount > 1 && !startDate && !timePeriodInDays) {
    localUris = scenes?.[i]?.localUris;
  }

  localUris.forEach((subItem) => {
    modelsAndTexturesArrayGenerator(subItem, models, textures, subItem.stable);
  });
}
```

Reply:
It's absolutely the right perspective.

Reply:
I split the if controls inside the for loop here and I think the code is more understandable now.

https://github.com/ikuseiGmbH/smart-village-app-app/pull/706/commits/ab256e92e32672bea38b55043e6cd5d8f428f5ca